### ISSUE TITLE
[Proposal] Search string for related models

### ISSUE THREAD

**Issue by mazedlx** [lorisleiva/laravel-search-string#6]
What do you think about adding the possibility to create a search string for related models e.g.

```php
class User extends Model 
{
    use SearchString;

    public function roles()
    {
        return $this->belongsToMany(Role::class);
    }
}
```

Using this search string 
```php
User::usingSearchString('active: 1 and role: admin');
```

would result in the following query
```php
User::join('role_user', 'role_user.user_id', '=', 'users.id')
    ->join('roles', 'roles.id', '=', 'role_user.role_id')
    ->where('users.active', '=', 1)
    ->where('roles.label', '=', 'admin')
    ->get();
```

**Comment by lorisleiva** [comment]
Hi there 👋 

Yeah, I'd love for that package to support every type of relationship.

I even want something like 
- `title: "My article" and comments > 3`
- `title: "My article" and comments` (equivalent to `comments > 0`)
- `title: "My article" and comments (author: johndoe) >= 1`
- `title: "My article" and category.name: Laravel`
- `title: "My article" and category (name: Laravel or name: Vuejs)`

It is quite a big task to do something like that properly. At the moment I don't have time to dedicate to this task but I'll leave that issue open until I do (or someone else does).

**Comment by mazedlx** [comment]
That'd be super awesome!

**Comment by lorisleiva** [comment]
Released in version [`v1.0.0`](https://github.com/lorisleiva/laravel-search-string/releases/tag/v1.0.0). 🎉
